### DISCUSSION TITLE
using AutoGluon.timeseries on regular series (not time series)

### DISCUSSION THREAD

**Discussion by inikishev** [autogluon/autogluon#4359] [Q&A] [answered]
I have a series which is basically just giant 1D numpy array, I was wondering if I could use autogluon.timeseries on it. This is what it says on wiki:
```
AutoGluon expects time series data in [long format](https://doc.dataiku.com/dss/latest/time-series/data-formatting.html#long-format). Each row of the data frame contains a single observation (timestep) of a single time series represented by

- unique ID of the time series ("item_id") as int or str

- timestamp of the observation ("timestamp") as a pandas.Timestamp or compatible format

- numeric value of the time series ("target")
```
surely there is no difference in using timestamp and using some normal series which is not a time series but something else, and just for example using id as time assuming points are evenly spaced.

I tried 
```
ids = list(range(series.shape[0]))
timestamps = [pandas.Timestamp(i) for i in ids]
```
and this is horribly slow, and didn't work. It says that I need to provide `freq` argument because it can't infer the frequency, but when I arbitrarily say `H` it tries to resample "from None to H", and I am pretty sure gets stuck because the data doesn't need resampling and is too big for that. I wonder if it can be skipped somehow because ids are all I really need

**Comment by shchur** [comment, marked as answer]
Hi @qq-me, here is how you can create a dataframe with dummy `item_id` and `timestamp` fields from a single series:
```python
import pandas as pd

def series_to_df(series: list, idx: int = 1) -> pd.DataFrame:
    return pd.DataFrame(
        {
            "item_id": [idx] * len(series),
            "timestamp": pd.date_range("1900-01-01", freq="YE", periods=len(series)),
            "target": series,
        }
    )

series = [1, 1, 2, 3, 5, 8, 13]  # list of numbers
df = series_to_df(series)
```

If you have multiple series, you can concatenate them into a single dataframe as follows:
```python
# List of lists
list_of_series = [
    [1, 2, 3, 4],
    [5, 2, 1, 2, 3, 5, 6],
]
combined_df = pd.concat([series_to_df(series, idx) for idx, series in enumerate(list_of_series)])
```

**Reply by inikishev** [reply]
> ```python
> len(series)
> ```

thank you, and I find that the reason mine didn't work is because I used `id = list(range(len(series)))`, why should it be `[1] * len(series)`?
